tools/bcdb_mock/pkg/directory/node_handlers.go

```diff
@@ -0,0 +1,242 @@
+		Hypervisor []string                                     `json:"hypervisor,omitempty"`
+	}{}
+
+	defer r.Body.Close()
+
+	if err := json.NewDecoder(r.Body).Decode(&x); err != nil {
+		return nil, mw.BadRequest(err)
+	}
+
+	nodeID := mux.Vars(r)["node_id"]
+	db := mw.Database(r)
+
+	if err := s.updateTotalCapacity(r.Context(), db, nodeID, x.Capacity); err != nil {
+		return nil, mw.NotFound(err)
+	}
+
+	if err := s.StoreProof(r.Context(), db, nodeID, x.DMI, x.Disks, x.Hypervisor); err != nil {
+		return nil, mw.Error(err)
+	}
+
+	return nil, nil
+}
+
+func (s *NodeAPI) registerIfaces(r *http.Request) (interface{}, mw.Response) {
+	log.Debug().Msg("network interfaces register request received")
+
+	defer r.Body.Close()
+
+	var input []generated.TfgridDirectoryNodeIface1
+	if err := json.NewDecoder(r.Body).Decode(&input); err != nil {
+		return nil, mw.BadRequest(err)
+	}
+
+	nodeID := mux.Vars(r)["node_id"]
+	db := mw.Database(r)
+	if err := s.SetInterfaces(r.Context(), db, nodeID, input); err != nil {
+		return nil, mw.Error(err)
+	}
+
+	return nil, mw.Created()
+}
+
+func (s *NodeAPI) configurePublic(r *http.Request) (interface{}, mw.Response) {
+	var iface types.PubIface
+
+	defer r.Body.Close()
+	if err := json.NewDecoder(r.Body).Decode(&iface); err != nil {
+		return nil, mw.BadRequest(err)
+	}
+
+	cfg := generated.TfgridDirectoryNodePublicIface1{
+		Gw4:    iface.GW4,
+		Gw6:    iface.GW6,
+		Ipv4:   iface.IPv4.ToSchema(),
+		Ipv6:   iface.IPv6.ToSchema(),
+		Master: iface.Master,
+		Type:   generated.TfgridDirectoryNodePublicIface1TypeMacvlan,
+	}
+
+	nodeID := mux.Vars(r)["node_id"]
+	db := mw.Database(r)
+	if err := s.SetPublicConfig(r.Context(), db, nodeID, cfg); err != nil {
+		return nil, mw.Error(err)
+	}
+
+	return nil, mw.Created()
+}
+
+func (s *NodeAPI) registerPorts(r *http.Request) (interface{}, mw.Response) {
+
+	defer r.Body.Close()
+
+	input := struct {
+		Ports []uint `json:"ports"`
+	}{}
+	if err := json.NewDecoder(r.Body).Decode(&input); err != nil {
+		return nil, mw.BadRequest(err)
+	}
+
+	log.Debug().Uints("ports", input.Ports).Msg("wireguard ports received")
+
+	nodeID := mux.Vars(r)["node_id"]
+	db := mw.Database(r)
+	if err := s.SetWGPorts(r.Context(), db, nodeID, input.Ports); err != nil {
+		return nil, mw.NotFound(err)
+	}
+
+	return nil, nil
+}
+
+func (s *NodeAPI) updateUptimeHandler(r *http.Request) (interface{}, mw.Response) {
+	defer r.Body.Close()
+
+	input := struct {
+		Uptime uint64
+	}{}
+	if err := json.NewDecoder(r.Body).Decode(&input); err != nil {
+		return nil, mw.BadRequest(err)
+	}
+
+	nodeID := mux.Vars(r)["node_id"]
+	db := mw.Database(r)
+	log.Debug().Str("node", nodeID).Uint64("uptime", input.Uptime).Msg("node uptime received")
+
+	if err := s.updateUptime(r.Context(), db, nodeID, int64(input.Uptime)); err != nil {
+		return nil, mw.NotFound(err)
+	}
+
+	return nil, nil
+}
+
+func (s *NodeAPI) updateUsedResources(r *http.Request) (interface{}, mw.Response) {
```

Comment:
This should be `updateReservedResources` we do not track "Used" resources, only total and reserved